/*
 * This file was automatically generated by EvoSuite
 */

package com.sap.netweaver.porta.ide.eclipse.server.control;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.evosuite.junit.EvoSuiteRunner;
import static org.junit.Assert.*;
import com.sap.netweaver.porta.core.AuthenticationReason;
import com.sap.netweaver.porta.core.AuthenticationRefusedException;
import com.sap.netweaver.porta.core.CoreException;
import com.sap.netweaver.porta.ide.eclipse.server.control.AuthCallback;
import org.junit.BeforeClass;

@RunWith(EvoSuiteRunner.class)
public class AuthCallbackEvoSuiteTest {

  @BeforeClass 
  public static void initEvoSuiteFramework(){ 
    org.evosuite.Properties.REPLACE_CALLS = true; 
  } 


  @Test
  public void test0()  throws Throwable  {
      AuthCallback authCallback0 = new AuthCallback(false, "mJr6Foc");
      AuthenticationReason authenticationReason0 = AuthenticationReason.INVALID_CREDENTIALS;
      // Undeclared exception!
      try {
        authCallback0.getCredentials(authenticationReason0);
        fail("Expecting exception: NoClassDefFoundError");
      } catch(NoClassDefFoundError e) {
        /*
         * Could not initialize class org.eclipse.swt.internal.wpf.COM
         */
      }
  }

  @Test
  public void test1()  throws Throwable  {
      AuthCallback authCallback0 = new AuthCallback(false, "tiSmetroR");
      AuthenticationReason authenticationReason0 = AuthenticationReason.AUTHORIZATION_REQUIRED;
      // Undeclared exception!
      try {
        authCallback0.getCredentials(authenticationReason0);
        fail("Expecting exception: NoClassDefFoundError");
      } catch(NoClassDefFoundError e) {
        /*
         * Could not initialize class org.eclipse.swt.internal.wpf.COM
         */
      }
  }

  @Test
  public void test2()  throws Throwable  {
      AuthCallback authCallback0 = new AuthCallback(false, "sw-Alirary.paWh");
      AuthenticationReason authenticationReason0 = AuthenticationReason.PERMISSION_DENIED;
      // Undeclared exception!
      try {
        authCallback0.getCredentials(authenticationReason0);
        fail("Expecting exception: NoClassDefFoundError");
      } catch(NoClassDefFoundError e) {
        /*
         * Could not initialize class org.eclipse.swt.internal.wpf.COM
         */
      }
  }

  @Test
  public void test3()  throws Throwable  {
      AuthCallback authCallback0 = new AuthCallback(true, "<h$J");
      AuthenticationReason authenticationReason0 = AuthenticationReason.AUTHORIZATION_REQUIRED;
      // Undeclared exception!
      try {
        authCallback0.getCredentials(authenticationReason0);
        fail("Expecting exception: NoClassDefFoundError");
      } catch(NoClassDefFoundError e) {
        /*
         * Could not initialize class org.eclipse.swt.internal.wpf.COM
         */
      }
  }
}
